File: src/main/java/com/upokecenter/rdf/ParserException.java
package com.upokecenter.rdf;
/*
Any copyright is dedicated to the Public Domain.
http://creativecommons.org/publicdomain/zero/1.0/
If you like this, you should donate to Peter O.
at: http://peteroupc.github.io/
 */

  /**
   * Exception thrown for errors that occur while parsing data. <p>This library
   * may throw exceptions of this type in certain cases, notably when
   * errors occur, and may supply messages to those exceptions (the message
   * can be accessed through the <code>Message</code> property in.NET or the
   * <code>getMessage()</code> method in Java). These messages are intended to be
   * read by humans to help diagnose the error (or other cause of the
   * exception); they are not intended to be parsed by computer programs,
   * and the exact text of the messages may change at any time between
   * versions of this library.</p>
   */

public final class ParserException extends RuntimeException {
private static final long serialVersionUID = 1L;
    /**
     * Initializes a new instance of the {@link
     * com.upokecenter.rdf.ParserException} class.
     */
    public ParserException() {
    }

    /**
     * Initializes a new instance of the {@link
     * com.upokecenter.rdf.ParserException} class.
     * @param message The parameter {@code message} is a text string.
     */
    public ParserException(String message) {
 super(message);
    }

    /**
     * Initializes a new instance of the {@link
     * com.upokecenter.rdf.ParserException} class. Uses the given message
     * and inner exception.
     * @param message The parameter {@code message} is a text string.
     * @param innerException The parameter {@code innerException} is an Exception
     * object.
     */
    public ParserException(String message, Throwable innerException) {
 super(message);
initCause(innerException);;
    }
  }

File: TurtleParser/PeterO/Rdf/ParserException.cs
/*
Any copyright to this work is released to the Public Domain.
In case this is not possible, this work is also
licensed under the Unlicense: https://unlicense.org/

 */
using System;

namespace PeterO.Rdf {
  /// <summary>Exception thrown for errors that occur while parsing data.
  /// <para>This library may throw exceptions of this type in certain
  /// cases, notably when errors occur, and may supply messages to those
  /// exceptions (the message can be accessed through the <c>Message</c>
  /// property in.NET or the <c>getMessage()</c> method in Java). These
  /// messages are intended to be read by humans to help diagnose the
  /// error (or other cause of the exception); they are not intended to
  /// be parsed by computer programs, and the exact text of the messages
  /// may change at any time between versions of this
  /// library.</para></summary>
  #if NET20 || NET40
  [Serializable]
  #endif
  public sealed class ParserException : Exception {
    /// <summary>Initializes a new instance of the
    /// <see cref='PeterO.Rdf.ParserException'/> class.</summary>
    public ParserException() {
    }

    /// <summary>Initializes a new instance of the
    /// <see cref='PeterO.Rdf.ParserException'/> class.</summary>
    /// <param name='message'>The parameter <paramref name='message'/> is a
    /// text string.</param>
    public ParserException(string message) : base(message) {
    }

    /// <summary>Initializes a new instance of the
    /// <see cref='PeterO.Rdf.ParserException'/> class. Uses the specified
    /// message and inner exception.</summary>
    /// <param name='message'>The parameter <paramref name='message'/> is a
    /// text string.</param>
    /// <param name='innerException'>The parameter <paramref
    /// name='innerException'/> is an Exception object.</param>
    public ParserException(string message, Exception innerException)
      : base(message, innerException) {
    }

    #if NET20 || NET40
    private ParserException(
      System.Runtime.Serialization.SerializationInfo info,
      System.Runtime.Serialization.StreamingContext context)
      : base(info, context) {
    }
    #endif
  }
}
